cli/src/main/java/org/jboss/sbomer/cli/feature/sbom/command/auto/GenerateEnvConfigCommand.java

```diff
@@ -0,0 +1,147 @@
+import picocli.CommandLine.Command;
+import picocli.CommandLine.Option;
+
+/**
+ * <p>
+ * Command to generate a runtime environment configuration file (compliant with SDKMan) for SBOMer generation
+ * </p>
+ */
+@Slf4j
+@Command(
+        mixinStandardHelpOptions = true,
+        name = "generate-env-config",
+        description = "Generates the runtime environment configuration used for automation for a given PNC build")
+public class GenerateEnvConfigCommand implements Callable<Integer> {
+
+    public static enum ConfigFormat {
+        yaml, json;
+    }
+
+    @Option(names = { "--build-id" }, required = true, description = "The PNC build identifier, example: AYHJRDPEUMYAC")
+    String buildId;
+
+    @Option(
+            names = { "--format" },
+            defaultValue = "yaml",
+            description = "Format of the generated environment configuration.")
+    ConfigFormat format;
+
+    @Option(
+            names = { "--target" },
+            paramLabel = "FILE",
+            description = "Location where the configuration environment file should be stored. If not provided, environment configuration will be printed to standard output.",
+            converter = PathConverter.class)
+    Path target;
+
+    @Inject
+    PncService pncService;
+
+    @Inject
+    ConfigReader configReader;
+
+    /**
+     * Retrieves environment configuration from a PNC Build, compliant with SDKMan.
+     *
+     * @return a Map object if the configuration could be retrieved or {@code null} otherwise.
+     */
+    private Map<String, String> environmentConfig() {
+        // Make sure there is no context
+        MDCUtils.removeContext();
+        MDCUtils.addBuildContext(buildId);
+        log.debug("Attempting to fetch environment configuration from a PNC build");
+
+        Build build = pncService.getBuild(buildId);
+
+        if (build == null) {
+            log.warn("Could not retrieve PNC build '{}'", buildId);
+            return Collections.emptyMap();
+        }
+        Map<String, String> buildEnvAttributes = build.getEnvironment().getAttributes();
+        if (buildEnvAttributes == null || buildEnvAttributes.isEmpty()) {
+            log.debug("Build environment attributes not found for the specified PNC build");
+            return Collections.emptyMap();
+        }
+        Map<String, String> envConfig = EnvironmentAttributesUtils.getSDKManCompliantAttributes(buildEnvAttributes);
+        if (envConfig == null || envConfig.isEmpty()) {
+            log.debug("No compliant SDKMan environment attributes could be generated");
+            return Collections.emptyMap();
+        }
+
+        return envConfig;
+    }
+
+    /**
+     *
+     * @return {@code 0} in case the environment config file was generated successfully. An empty Map is valid as
+     *         default versions will be used
+     */
+    @Override
+    public Integer call() throws Exception {
+        Map<String, String> envConfig = environmentConfig();
```

Comment:
Maybe we could add a check whether the build exists and fail the execution. Non-existing build and the inability to fetch environment configuration are treated the same by the code.

Reply:
good point, changed the above!